src/AuthenticatedAPIClient/authInterface.js

```diff
@@ -101,8 +105,35 @@ export default function applyAuthInterface(httpClient, authConfig) {
       // Attempt to refresh the JWT cookies.
       return httpClient.refreshAccessToken()
         // Successfully refreshed the JWT cookies, fire the callback function.
-        .then(() => {
-          callback(httpClient.getDecodedAccessToken());
+        .then((response) => {
+          const refreshedAccessToken = httpClient.getDecodedAccessToken();
+
+          /* istanbul ignore next */
+          if (refreshedAccessToken === null) {
+            // This is the success block for refreshing an access token.
+            // Sometimes the access token is null for an unknown reason.
+            // Log here to learn more.
+            if (httpClient.loggingService && httpClient.loggingService.logError) {
+              httpClient.loggingService.logError(
+                new Error('Access token is null after refresh.'),
+                {
+                  previousAccessToken: accessToken,
+                  axiosResponse: response,
+                },
+              );
+
+              // Wait 50ms and check again. Maybe we're in a race to set the cookie?
+              setTimeout(() => {
+                const checkAccessToken = httpClient.getDecodedAccessToken();
+                if (checkAccessToken !== null) {
+                  const unexpectedTokenError = new Error('Access token was null after refresh and now is not 50ms later');
+                  httpClient.loggingService.logError(unexpectedTokenError);
+                }
+              }, 50);
+            }
+          }
+
+          callback(refreshedAccessToken);
```

Comment:
Maybe add a TODO comment?  We need to determine what to do here.  Do we really ever want to call the callback function with a blank refreshedAccessToken?